internal/services/producer.go

```diff
@@ -9,12 +9,32 @@ package services
 import (
 	"log"
 	"os"
+	"strconv"
 	"time"
 
 	"github.com/Shopify/sarama"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 	"github.com/strimzi/strimzi-canary/internal/config"
 )
 
+var (
+	recordsProduced = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name:      "records_produced_total",
+		Namespace: "strimzi_canary",
+		Help:      "The total number of records produced",
+	}, []string{"clientid", "partition"})
+
+	recordsProducedFailed = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name:      "records_produced_failed_total",
+		Namespace: "strimzi_canary",
+		Help:      "The total number of records failed to produce",
+	}, []string{"clientid", "partition"})
```

Comment:
If we end up with a lot of these across both producer and consumer, it might be nice to consider a factory or separate package or similar to gather them into one place long-term, but that's a nice-to-have outside of scope of PR.

Reply:
What's the natural golang pattern here? Having a canary_metrics package with just these vars declaration?

Reply:
Hmm, tempted to suggest just that, maybe a map[string]CounterVec or some other wrapping entity in a new pkg...not 100% sure. Probably worth leaving as-is and seeing how it works with testing/mocking and not making an issue out of it until it's necessary.

Reply:
Ok so if the PR is fine, I guess you can approve it ;)